test/config/config.test.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+const test = require('tape')
+
+const { CONFIG_DEFINITIONS } = require('./fixtures/config-schema')
+
+const Agent = require('../../lib/agent')
+const { NoopTransport } = require('../../lib/noop-transport')
+const MockLogger = require('./_mock_logger')
+
+// Options to pass to `agent.start()` to turn off some default agent behavior
+// that is unhelpful for these tests.
+const agentOpts = {
+  centralConfig: false,
+  captureExceptions: false,
+  metricsInterval: '0s',
+  cloudProvider: 'none',
+  logLevel: 'warn'
+}
+const agentOptsNoTransportNoLogger = Object.assign(
+  {},
+  agentOpts,
+  {
+    logger: new MockLogger(),
+    transport: function createNoopTransport () {
+      // Avoid accidentally trying to send data to an APM server.
+      return new NoopTransport()
+    }
+  }
+)
+
+function hasType (def, type) {
+  return def.types && def.types.indexOf(type) !== -1
+}
+
+function testEnvVar (t, option, value1, value2) {
```

Comment:
nit: Perhaps calling this `function assertEnvVar` or similar, which is more of a current naming pattern:

```
% rg 'function test' test | wc -l
       5
% rg 'function assert' test | wc -l
      50
```